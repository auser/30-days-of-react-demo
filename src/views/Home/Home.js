import React from 'react';
import {connect} from 'react-redux';

export const Home = (props) => {
  const {currentTime, actions} = props;

  return (
    <div className="content">
      <h1>Welcome</h1>
      <p>The time is {currentTime.toString()}</p>
      <button onClick={() => actions.currentTime.updateTime({})}>
        Update time
      </button>
    </div>
  );
}

export default connect(state => ({
  currentTime: state.currentTime.currentTime
}))(Home);
