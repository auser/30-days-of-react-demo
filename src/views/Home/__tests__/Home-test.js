/* eslint-disable */
import React from 'react';
import renderer from 'react-test-renderer';
import sinon from 'sinon';

jest.unmock('../Home');
import {Home} from '../Home';

const identity = () => {};
let actions = {
  currentTime: { updateTime: identity }
}
let currentTime = { currentTime: new Date() };

describe('Home', function() {
  beforeEach(() => {
    actions.currentTime.updateTime = sinon.spy();
  })

  it('renders a logout view', () => {
    const component = renderer.create(
      <Home actions={actions} currentTime={currentTime} />
    )
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
    expect(actions.currentTime.updateTime.called).toBeFalsy();

    // set the login input
    const button = tree.children[tree.children.length - 1];
    button.props.onClick();
    expect(actions.currentTime.updateTime.called).toBeTruthy();
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
