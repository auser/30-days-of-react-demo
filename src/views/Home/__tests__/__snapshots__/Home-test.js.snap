exports[`Home renders a logout view 1`] = `
<div
  className="content">
  <h1>
    Welcome home!
  </h1>
  <p>
    The time is 
    [object Object]
  </p>
  <button
    onClick={[Function onClick]}>
    Update time
  </button>
</div>
`;

exports[`Home renders a logout view 2`] = `
<div
  className="content">
  <h1>
    Welcome home!
  </h1>
  <p>
    The time is 
    [object Object]
  </p>
  <button
    onClick={[Function onClick]}>
    Update time
  </button>
</div>
`;
